### DISCUSSION TITLE
License of the project

### DISCUSSION THREAD

**Discussion by fbartels** [winapps-org/winapps#11] [General]
Hi,

not to be the bearer of bad news, but I noticed that with pulling in the original project code, the same commit also adds a GPL license file:

- https://github.com/winapps-org/winapps/commit/c6f3c3443276b5304af11589df0cb17ba59c6622

While the original author expressed in https://github.com/Fmstrat/winapps/issues/17#issuecomment-724025702 that he would like to add a gpl license, one cannot simply apply an oss license to previously unlicensed code. A license change would require permission from all previous contributors (and when pulling in pull requests to the original project also the permission from the authors of these).

**Comment by oskardotglobal** [comment]
Wait, but if the code isn't licensed at all, wouldn't that mean its all rights reserved for fmstrat & all contributors? So we wouldn't be able to do anything?

**Reply by fbartels** [reply]
Yes, although some of the commits from other contributors could be classified as trivial and may therefore not require approval of the license change. But I am not a lawyer, so take this with a grain of salt.

**Reply by oskardotglobal** [reply]
But if the code is all rights reserved, that would disallow us forking the project

**Reply by fbartels** [reply]
Github does not take this into account and therefore does not actively prevent forks being created.

But on one hand there is the license issue and on the other hand there is an original author who does not seem to enforce license violations and even merged third party modifications.

**Comment by oskardotglobal** [comment]
I have removed the license for now. See https://github.com/winapps-org/winapps/commit/6c57c2e13b2c6b96331e4c667b3db4ad4e670381

**Comment by freechelmi** [comment]
Sorry My fault when I created the repo  , I'll be more cautious next time

**Comment by LDprg** [comment]
So we may not be able to licence the legacy code base? That would also disalow redistribution over some package manager.

**Reply by oskardotglobal** [reply]
Well kind of, but not like fmstrat or contributors are going to do anything

**Reply by LDprg** [reply]
That's true, but the licence missing was a problem in the past too. 
Like https://bbs.archlinux.org/viewtopic.php?id=267827

**Comment by oskardotglobal** [comment]
Ok, we've kind of got a problem. This fork along with all others is very much violating copyright law and as @LDprg mentioned, this along with having no license will be problematic if we try to get this project onto any package manager. We need written permission from fmstrat & all contributors in order to be able to relicense this project. 

The code is probably all rights reserved, as we can assume fmstrat is american. If he wasn't, then other copyright laws would apply which may or may not make the code all rights reserved. 
From his website: 
> As a member of the leadership team for a medical technology company based out of NY ..

However, there's only 8 other contributors besides people in this org. If these commits are non-trivial (which most of them are), we can just purge them from the history, which would limit the amount of CLAs to be signed by a lot. I'll take a look at this now

**Reply by oskardotglobal** [reply]
Ok, all the commits from the 8 contributors were non-trivial and I've dropped out all of them. Had to do some nasty rebasing and force-pushing, but we now just require fmstrat's approval, then we can relicense the project.

**Reply by freechelmi** [reply]
Great work !

**Reply by LDprg** [reply]
Who is contacting fmstrat? If he approves the relicensing he could change the searching for maintainer notice to reference to our repo. This would really help in terms of contributions, etc.

**Reply by oskardotglobal** [reply]
I'd prefer for fmstrat to sign the cla via cla-assistant once thats set up, but once thats done, I can try to contact them

**Comment by oskardotglobal** [comment]
Now we just need the CLA for people to sign. We could use something like https://contributoragreements.org/ca-cla-chooser to create one and https://cla-assistant.io/ to make people automatically sign it on PRs.

@freechelmi
It would probably be best if would create it, since you created the org. (I'm not asking for you to be a legal representative, if we get in trouble we just all bail)

**Reply by freechelmi** [reply]
OK we'll look into it tonight ( GMT )

**Reply by LDprg** [reply]
Is the CLA stuff ready @freechelmi ?

**Comment by oskardotglobal** [comment]
See #15

**Comment by oskardotglobal** [comment]
By the way, all of you can go ahead and sign the cla too: https://cla-assistant.io/winapps-org/winapps

**Reply by fbartels** [reply]
Done

**Reply by LDprg** [reply]
Done

**Reply by freechelmi** [reply]
Done , thanks

**Comment by oskardotglobal** [comment]
I just realised: We can already start merging PRs if the CLA is signed by the person. The CLA people sign permits relicensing, so once #15 is merged the code will be GPL. Code merged now is just All rights reserved for winapps-org.

In the CLA:
> [...] with the right to transfer an unlimited number of non-exclusive licenses or to grant sublicenses to third parties

**Comment by LDprg** [comment]
@oskardotglobal Any response from fmstrat regarding the cla?

**Comment by oskardotglobal** [comment]
Nope, haven't heard back yet. Should I ask again?

**Reply by LDprg** [reply]
Yeah, maybe he forgot about it. However you may check if he has signed the cla already (if possible).

**Reply by oskardotglobal** [reply]
I asked again and of course checked beforehand if he already did. He did not.

**Reply by LDprg** [reply]
@oskardotglobal any news on this one?

**Reply by oskardotglobal** [reply]
Nope ;-;
